src/main/java/com/bunq/sdk/json/BigDecimalTypeAdapter.java

```diff
@@ -29,7 +29,11 @@ public void write(JsonWriter output, BigDecimal value) throws IOException {
 
   @Override
   public BigDecimal read(JsonReader input) throws IOException {
-    if (input.peek() == JsonToken.NUMBER) {
+    JsonToken type = input.peek();
+
+    if (type == JsonToken.NUMBER) {
+      return new BigDecimal(input.nextInt());
```

Comment:
Why nextInt instead of double? Int will not return you the fractions.

Reply:
😂 🤦‍♂️